src/orange/OrangeTrackView.hh

```diff
@@ -203,13 +204,31 @@ OrangeTrackView::operator=(const Initializer_t& init)
     local.temp_sense = this->make_temp_sense();
 
     // Initialize logical state
-    auto tracker = this->make_tracker(UniverseId{0});
-    auto tinit   = tracker.initialize(local);
-    // TODO: error correction/graceful failure if initialiation failured
-    CELER_ASSERT(tinit.volume && !tinit.surface);
+    size_type uid = 0;
+
+    detail::UnitIndexer unit_indexer(params_.unit_indexer_data);
+
+    while (true)
+    {
+        auto tracker = this->make_tracker(UniverseId{uid});
+        auto tinit   = tracker.initialize(local);
+        // TODO: error correction/graceful failure if initialiation failured
+        CELER_ASSERT(tinit.volume && !tinit.surface);
+
+        OpaqueId<VolumeRecord> vrid(tinit.volume.unchecked_get());
 
-    // Save local data
-    states_.vol[thread_] = tinit.volume;
+        auto daughter_id = params_.volume_records[vrid].daughter;
+
+        if (daughter_id)
+        {
+            uid = daughter_id.unchecked_get();
+            continue;
+        }
+
+        states_.vol[thread_]
+            = unit_indexer.global_volume(UniverseId{uid}, tinit.volume);
+        break;
+    }
```

Comment:
`while (true) { ... continue; ... break; }` is not a natural way for code to flow: that's basically a "goto". Code is easier to read and understand if there's exactly one place the loop decision is made, and when that's in the beginning or end of the loop.

Try this instead:
```c++
    UniverseId uid{0};
    VolumeId volid{};
    do
    {
        auto tracker = this->make_tracker(uid);
        auto tinit   = tracker.initialize(local);
        // TODO: error correction/graceful failure if initialiation failed
        CELER_ASSERT(tinit.volume && !tinit.surface);
        volid = tinit.volume;

        // WRONG: volume returned by tinit is local, 
        OpaqueId<VolumeRecord> vrid(volid.unchecked_get());
        uid = params_.volume_records[volid].daughter;
    } while (!uid);

    detail::UnitIndexer unit_indexer(params_.unit_indexer_data);
    states_.vol[thread_] = unit_indexer.global_volume(uid, volid);
```

Reply:
I think  the `volid` in 

`uid = params_.volume_records[volid].daughter;`

needs to be `vrid`.

Reply:
(and the while condition should be simply `while (uid)`)